### PR TITLE
Export/backup of large number of attachments

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
This fix should help reduce the CPU load of backing up apps that contain a large number of exports, as well as allowing import of these larger apps in self host environments. There will be some configuration required to achieve this, namely:

The `budibase/proxy` image would need to be overridden, by default the max request size it allows is 1000MB - this needs to be raised to allow apps of the GB size. Note that I have increased the development size to 50GB in `nginx.dev.conf` - this will allow us to test these massive export/imports in development, but normal cloud/self host environments won't have this option/will need to update it.

I've added a new environment variable, `MAX_IMPORT_SIZE_MB` - which can be used to set the maximum size of file which can be imported to the `budibase/apps` or server instance - by default this is usually 100MB.

Addresses: https://linear.app/budibase/issue/BUDI-7586/customers-app-backups-are-failing-in-cloud

### Testing
A note on testing, there are some test cases which already cover some parts of this (e.g. the `backup.spec.ts` test cases), but in our test environment we don't really have MinIO access today. This is a situation where we really need a more complex test bench which we can use a full environment to test for scaling problems, like re-using a massive exported file to create an app and check that it still works.

### CODE REVIEW

File: packages/backend-core/src/utils/utils.ts [hunk 1]
```diff
@@ -159,28 +155,24 @@ export function isValidInternalAPIKey(apiKey: string) {
...
+export function getCookie<T>(ctx: Ctx, name: string) {
   const cookie = ctx.cookies.get(name)
 
   if (!cookie) {
-    return cookie
+    return undefined
   }
 
-  return openJwt(cookie)
+  return openJwt(cookie) as T
```
Comment (samwho) [inline packages/backend-core/src/utils/utils.ts:175]: How come this over `openJwt<T>(cookie)`?

File: packages/worker/src/tests/TestConfiguration.ts [hunk 2]
```diff
@@ -209,7 +210,7 @@ class TestConfiguration {
       sessionId: "sessionid",
       tenantId: user.tenantId,
     }
-    const authCookie = auth.jwt.sign(authToken, coreEnv.JWT_SECRET)
+    const authCookie = jwt.sign(authToken, coreEnv.JWT_SECRET as Secret)
```
Comment (samwho) [inline packages/worker/src/tests/TestConfiguration.ts:213]: On a few occasions I have wished for better typing on our environment variables. Might noodle with that one lab day.

File: packages/server/src/sdk/app/backups/exports.ts [hunk 3]
```diff
@@ -150,19 +151,21 @@ export async function exportApp(appId: string, config?: ExportOpts) {
...
-
-      fs.rmSync(path)
+      // skip the attachments - too big to encrypt
+      if (file !== ATTACHMENT_DIRECTORY) {
+        await encryption.encryptFile(
+          { dir: tmpPath, filename: file },
+          config.encryptPassword
+        )
+        fs.rmSync(path)
+      }
```
[the remark this answers is not in the document]
Comment (mike12345567) [reply packages/server/src/sdk/app/backups/exports.ts:161]: we encrypt the contents of the database, as it includes schemas of databases, data etc - there could be information in here which is private.

The same is true for attachments, but we simply cannot encrypt them as they can get too big, the fact we try to do this today is likely part of the performance issues we occasionally have in the cloud - this should help manage this going forward.

We already make it quite clear that app exports should be looked after and not freely shared if they contain important information, but specifically we wanted to make sure that very private data, like say REST queries which could include API keys, was properly secured. The attachments were a bit of a by-product but probably never shouldn't have been encrypted.

File: packages/worker/src/tests/TestConfiguration.ts (hunk 2 above)
Comment (mike12345567) [reply packages/worker/src/tests/TestConfiguration.ts:213]: Yeah agreed, theres definitely more we could do with this.

File: hosting/nginx.dev.conf [hunk 4]
```diff
@@ -42,7 +42,7 @@ http {
   server { 
     listen       10000 default_server;
     server_name  _;
-    client_max_body_size 1000m;
+    client_max_body_size 50000m;
```
[the remark this answers is not in the document]
Comment (mike12345567) [reply hosting/nginx.dev.conf:45]: No this has been discussed, for now we have decided this is just an un-acceptable option in production environments. Self hosters can change it by updating their proxy image, but wanted to be able to mess with these sort of exports without issue locally.

File: packages/server/src/sdk/app/backups/exports.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (samwho) [reply packages/server/src/sdk/app/backups/exports.ts:161]: Will this change require communication with customers? Will it break existing exports?

[the remark this answers is not in the document]
Comment (mike12345567) [reply packages/server/src/sdk/app/backups/exports.ts:161]: I'll give that a test - it shouldn't break them, but I'll confirm that, if it does then I'll adjust for it.

**Review by samwho — APPROVED** [review]

**Pull request merged**